File: Java/Lista3/Zad3/prob.java
public class prob{

    public Boolean proba(String z1)
    {
        int m=0;
        try
        {
        m=Integer.parseInt(z1);
        if(m<=0)
            throw new Small();
        }
        catch(NumberFormatException e)
        {
            System.out.println(z1 + " - nie jest liczbą");
            return false;
        }
        catch(Small e)
        {
            System.out.println(z1 + " - nie jest dodatnie");
            return false;
        }
        catch(Exception e)
        {
            System.out.println(z1 + " - jest nie odpowiednie");
            return false;
        }
        return true;
    }
}

File: Cpp/Lista3/Zad2/prob.cpp
#include "hed.hpp"
#include "Fig.hpp"
#include <string> 
bool proba(string z1)
{
    int m=0;
    try
    {
        m=stoi(z1);
        if(m<=0)
        throw Small();
    }
    catch(std::invalid_argument e)
    {
        cout<<z1 << " - nie jest liczbą"<<endl;
        return false;
    }
    catch(Small e)
    {
        cout<<z1 + " - nie jest dodatnie"<<endl;
        return false;
    }
    catch(exception e)
    {
        cout<<z1 << " - jest nie odpowiednie"<<endl;
        return false;
    }
    return true;
}
